### ISSUE TITLE
[REACT] How to manually/dynamically play & stop particle emitters

### ISSUE THREAD

**Issue by ghost** [tsparticles/tsparticles#1184]
How can i manually play and stop the particle emitters in React.js?

Can't find anything on the docs..

**Which version are you using?**
1.18.11

**Is the latest version affected?**

- [ x] Yes
- [ ] No

**Which library are you using?**

- [ ] Vanilla JS (`tsparticles`)
- [ x] React (`react-tsparticles`)
- [ ] Vue.js 2.x (`particles.vue`)
- [ ] Vue.js 3.x (`particles.vue3`)
- [ ] Angular (`ng-particles`)
- [ ] Svelte (`svelte-particles`)
- [ ] Preact (`preact-particles`)
- [ ] Inferno (`inferno-particles`)
- [ ] jQuery (`jquery-particles`)

**Comment by matteobruni** [comment]
Hello @lemurweb-dev,

This is a great question, I never thought about that before but I found an easy solution (until I make it simpler).

After retrieving the `container` using the `ref` object you can use this snippet to manage the emitter:

```javascript
container.plugins.get("emitters").array[0].play();
container.plugins.get("emitters").array[0].pause();
```

Obviously `array[0]` can be replaced with any valid index.

If you don't know how to retrieve the `container` checkout this: #1108

**Comment by ghost** [comment]
Thank you Matteo,
i managed to get the Ref of the container and i have no errors when calling the
```javascript
containerRef.current.plugins.get("emitters").array[0].play();
``` 
function, but the emitter doesn't emit still.
I'm using tsParticles to have some kind of "rocket launch" smoke effect played only when user presses a button. So i set
`autoPlay: false,` in the options.
This is my button pressed function:

```javascript
const homeButtonClickHandler = () => {
    containerRef.current.plugins.get("emitters").array[0].play();
    setTimeout(function () {
      containerRef.current.plugins.get("emitters").array[0].pause();
    }, 2000);
  };
```

I managed to get it to work by calling the global container play function instead, like this:
```javascript
const homeButtonClickHandler = () => {
    containerRef.current.play();
    setTimeout(function () {
      containerRef.current.stop();
    }, 2000);
  };
```
But when i stop it the particles disappear or if i pause it the particles remain still.
What i want is to emit particles for 2 seconds, then stop emitting them and the active particles still on screen should move out of the screen (i set `direction: "top",` and `speed: 15,`).

Can you help me out?
Thanks

**Comment by matteobruni** [comment]
You need to start the container before emitters if you disable the `autoPlay`. The container start will initialize it correctly. If you just want emitted particles you can set the general `particles.number` to `0` and you'll have only the emitted ones.

Hope it helps, if you can share the config I can help more.

**Comment by Anku5hk** [comment]
I found a workaround, so if you want to emit for 3 seconds, you start with `start()` and `play()`, then pause the emitter at 3 second with `containerRef.current.plugins.get("emitters").array[0].pause();` and then wait for particles to settle down for say 9 seconds, then finally call `containerRef.current.stop();` here's the code

    containerRef.current.start();
    containerRef.current.play();
    setTimeout(function () {
      containerRef.current.plugins.get("emitters").array[0].pause();
    }, 3000);
    setTimeout(function () {
      containerRef.current.stop();
    }, 9000);

**Comment by titusdecali** [comment]
Honestly, that is far too complex for quite basic functionality. Perhaps solve this with props like `pauseDelay`(how long till items fade away after pausing), `play` (bool), `pause`(bool), `stop`(bool) would make for a much nicer interface.

**Comment by matteobruni** [comment]
> I found a workaround, so if you want to emit for 3 seconds, you start with `start()` and `play()`, then pause the emitter at 3 second with `containerRef.current.plugins.get("emitters").array[0].pause();` and then wait for particles to settle down for say 9 seconds, then finally call `containerRef.current.stop();` here's the code
> 
> ```
> containerRef.current.start();
> containerRef.current.play();
> setTimeout(function () {
>   containerRef.current.plugins.get("emitters").array[0].pause();
> }, 3000);
> setTimeout(function () {
>   containerRef.current.stop();
> }, 9000);
> ```

Or you can just use the options like in this sample: https://codepen.io/matteobruni/pen/xxjKgPg

Setting the `emitters.life` options like this: 

```
    life: {
      count: 1,
      duration: 3,
      delay: 0
    },
```

and the `duration` field in the main options with a milliseconds value (it's a bug that will be fixed in #4605, that value should be seconds)

```
duration: 9000
```

And you'll have the same without having breaking changes in future releases.
